### ISSUE TITLE
Idea on how to handle the concept of removing solo antag objectives

### ISSUE THREAD

**Issue by kilozombie** [d3athrow/vgstation13#19489]
I want this wild ride to be over before it even begins. I want this to not be another controversy and for us to all come to a good conclusion. In response to https://github.com/vgstation-coders/vgstation13/pull/19485 and https://github.com/vgstation-coders/vgstation13/pull/19487, here is my idea on how to have us all come to a good conclusion.

1. Make the PR serious so that the coders take it seriously.
2. To begin with, make it something that admins must manually toggle on at roundstart, regardless of if that round is solo antags or not.
3. After admins have tested it on a few dozen rounds over a few weeks with adjustments and edits, make a server poll.
4. If the poll goes in favor of removing solo antag objectives, merge it.

This issue is not an invitation for shitflinging from anyone or towards anyone. Love you all. Thanks

**Comment by Inorien** [comment]
I don't really like the idea of rolling antag and using that as an excuse to do whatever the fuck you want. We frown upon the department head who takes it upon themselves to be security, so why should antag objectives be different? The only objective that runs counter to the excuse to murderbone that seems to be at the heart of this whole thing is "minimise casualties", and no one knows how that even works. 
I think its a problem when someone gets an objective to, say, steal the captain's laser, but then makes zero effort to do so and instead goes on a murder rampage. It'd be better if we had the culture instead of "do whatever you want as long as you achieve your objective", at the same time only generating objectives that will actually be challenging to do. For example, "steal the captain's laser" can't be rolled if there is no captain; "steal a hypospray" can't be rolled if no CMO. Otherwise, its just a matter of breaking into an empty office and the challenge is already over. 
People seem to dislike the current monotony of objectives, perhaps we should find a way to encourage more interaction and engagement through use of more "advanced" objectives than just caving in and saying "sure, become a murderous rampaging lunatic if you want". Even Cult is somehow able to complete its objectives and theres no culture of mindlessly murdering everyone and ignoring objectives there. Same with nukeops, they always make an effort to do the nuke thing, rather than just ignoring the nuke and spending the round shooting scientists.

**Comment by kilozombie** [comment]
Nothing has been done except hypothesize about why this would be a bad change. Here is the only point of reference we can cite:

- Bay did it a long time ago.

I think testing it with actual players on the actual server in an actual round will tell us a lot more than circular discussion about how it would/wouldn't destroy the game forever.

**Comment by Inorien** [comment]
I'd still rather replace it with something else rather than simply getting rid of objectives and granting a blank check to whoever happens to get the roll. Clearly there's a problem with current objectives, if something can be done to fix this problem and improve the feature then that would be preferable, ideally in a way that encourages more interaction with the round beyond killing anything that moves or dude bombs

**Comment by kilozombie** [comment]
> simply getting rid of objectives and granting a blank check to whoever happens to get the roll

I mean, getting solo antag is already a blank check. It's just a blank check that pretends it's not a blank check. Players who get antag get at least a _pointed_ direction towards doing certain things, most of which are not entirely interesting in a vacuum. I understand the argument for making better objectives, but I also think that our current system is fundamentally flawed because it makes antagging into its own minigame, wherein the fastest and most convenient route is also the most boring.
Any objective you give except "be yourself" or "do whatever you think the Syndicate wants you to do" can be gamed and reduced into an extremely boring experience, IMO.

**Comment by Inorien** [comment]
I don't disagree one bit, it lies in the interpretation of solo antag. Removing objectives makes it an explicitly blank check, though, and as we know with purged AI or borgs (inb4 muh convicted criminal brains) a literal blank check leads to fun for only the person with the check. The only actual blank check antag we have is wizard, and we all know what that's like. I think the core of the issue is that objectives have no incentive and yield nothing interesting for the antag from a gameplay perspective (see: steal a thing from an empty office) and less of "lul rules are made to be broken the game cant tell me what to do". Ideally we'd have a higher pop to encourage more player interaction through use of objectives, and related to that I think shit antags who use their blank check as an excuse to murderbone drive away such populations through stealing the round enjoyment for themselves. Do you think that could be a contributing factor?

**Comment by kilozombie** [comment]
It's hard to say. I do think removing objectives would increase murderbone, but giving a loose objective without a winstate might not. I think greentext might be the problem more than objectives themselves, but I feel like antags would have more creative ideas if the playing field was already equipped for them to do so.

Okay, here's an idea. What if we treated objectives less like "objectives" and more like writing prompts, and applied the same philosophy that we do for writing prompts? Something that gives an antag a direction, but encourages them to find their own flair. Objectives currently sort of try to get to this, there are obviously multiple ways to kill somebody or steal something, but they often fall flat when you take into account that the winstate is still to _do the thing_, not to do it in an interesting way.

What about an objective like, "Cause rebellion", or "Give somebody a nightmarish shift", or "Make this a night to remember for [person]." or something?

**Comment by yclatious** [comment]
If we want vague prompts, why not something like this for generalized, non concrete objectives?

"You are a member of the Syndicate and employed by them to cause chaos and havoc upon the station. Sabotage, kill and betray to cause trouble for the station and its denizens, using whatever means you like."

or

"As a member of the Syndicate, you have been given no specific tasks, but the goal remains clear. Cause chaos, mayhem and destruction upon the station, be it by killing crew, sabotaging systems, making the station a unliving hell or whatever else you can achieve."

Or, if youd like to have pseudo objectives:

Shitting on someone the whole shift:
-"The Syndicates reputation needs to be strengthened. Find someone or a group of crewmen and make their lives absolute hell, as to establish the Syndicate as a strong threat that can strike anywhere, even at the heart of a enemy station."

Assasination of a guy/group:
-"People have begun to forget the strength of the Syndicate. Assasinate atleast one member of the station you are on, and make sure they cannot be recovered. We must establish ourselves as a rutheless organization who can and will kill with no regrets." 

Theft:
-"The Syndicate requires money, materials and technology to further the war on NT. Acumulate as much powerfull tech, rare minerals and money as possible and bring them back to fuel our activities."

Just a quick bunch of prompts. Personaly, I think objectives are really meh, but people are naturaly shit if they dont wanna do good gimmicks or just cant be assed to. Objectives are crutches, you, me, we can all do a great gimmick and be great antags, we just need to give half a ass to be good and make it interesting or atleast a situation hard to fix and making it engaging for others.

Id like more antag interaction amongst eatchother, but thats a different thing. I stand by us all being able to naturaly do more than just theft and objectives hampering us down from our creativity and just indulging our lazyness, but thats up to the group as a whole to agree or disagree. 

Personaly however, I can agree with one thing. "Just bee yourself" is fucking shit and a meme. Fix it with a actual ic prompt like mine or something akin to it.

**Comment by SonixApache** [comment]
there's people that ignore objectives and do things that make a round fun for everyone
there's people that ignore objectives and make a round boring for everyone 
there's people that do objectives and make the round fun for everyone
there's people that do objectives and make the round boring for everyone

There's no real cohesion, and wide-combing them isn't really going to achieve anything.
I think the "WE'RE NOT BAY!!1" argument people throw, from both sides, is absolutely retarded and unrelated.
Yclat's probably got the right idea, replacing objectives with general guidelines (or god forbid, straight-up ideas) for our antags. Personally, I think there should just be a bit more guidance for antaggery. Punish lowpop murderboners and dudebombers, support interesting gimmicks.

**Comment by PJB3005** [comment]
This is the kinda thing that's just "literally fucking what" on first sight, but honestly, I kinda do agree with it.

Antag objectives are fucking shit. The only objective that can ACTUALLY be interesting if played out is Hijack, because it actually has a lot of effect on the round on its own. All the other objectives are too focused on specific things to be enjoyable for *anybody* if followed strictly. Steal/assassinate involves like 1 or 2 people at most, and assassinate if done effectively is just parapen and round ruined for the target.

Yeah, you *can* make some objectives interesting, but the kinds of people to actually do that are the kinds of people who have a hell of a lot of potential with the more prompt-style thing I've seen suggested and approve of.

Prompts also solve the lack of direction from not having objectives at all.

That said, I do want a reminder to the players that traitors should try to make the round interesting for everybody involved, not just themselves.

**Comment by Exxion** [comment]
Not that I should really weigh in, since even while I still played I had nearly all antag roles disabled, but I generally like objectives in theory, but not in practice
Can't speak for anyone else, but having some concrete goal is important to me. But if said concrete goal is "kill person A and also yourself," that can be done by a vegetable in under five minutes, ruining two people's rounds and not improving anyone's.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Why objectives are bad:
1) They rely on a binary fail/succeed which limits them severely because more complex objectives would be a nightmare/impossible for the code to check for failure/completion. Example: Minimize casualties is 100% impossible to code in satisfactorily, and so would be anything with a broad scope or anything that requires quantifying player interaction which is 90% of the game.
2) They have no awareness of what's going on in the round. You can have the most interesting bus/gimmick/court case going on and the code will never have any hope of detecting it, it will just pick a random thing to do. It's entirely fair to say that objectives are an entirely separate minigame running on top of the real game.
3) They're extremely concrete with a narrow scope (mostly because of point 1). Because of this it's only natural for the player to focus inside this narrow scope to get the result they need in a simple/safe way. Compare "steal X item" to "steal from X department": In the former the player will probably focus on stealing that specific item and then consider it done and fuck off. In the latter the player will start to think what would be the most interesting/helpful thing to steal, and after they get it, they have a motivation to keep stealing. A broader scope greatly improves how the task gets approached.

How to improve them:
1) Just ditch the success/fail conditions. You don't need extremely complex and finicky code to determine if you really made the RD mad or not. Do you feel like you won? Then good, you won.
2) Write them to be more of a motivation than an objective. "Kill the CMO" = bad, "The CMO ruined your life" = good because there's a million ways to go about getting your revenge.
3) Make it common for players and admins to make new objectives mid-game. They'll naturally try to pick something they think will work based on the current round. Really, you should be able to make your own objectives if you wanted.
4) At roundstart give players the choice between 3 different objectives, if you get an objective you don't care about it that you know wouldn't be interesting because of playercount/manifest/whatever, you should be able to reroll it, why not?

**Comment by MadmanMartian** [comment]
Remove the objective greentext, instead have a stats thing

- Unique stats for that antag (Vampire blood count, changeling genome count, etc)
- If they have high-value items in their possession (Nuke disk, Ians collar, etc)
 - Killcount

And then have objectives be completable mid-round to help the antag out. (Smuggle out some high-value item to the syndicate: You get some telecrystals)

**Comment by Blithering** [comment]
Maybe have objectives be an optional thing with some purpose to them, for example a traitor presses a 'request additional task' button on his pda, it creates a prompt for admins to fill in a custom objective and possibly mail him more tcs to achieve it, if no admin responds it generates a standard objective. At that point the traitor is watched by the syndicate and has to put some effort into getting dat sweet greentext or risk the wrath of admins pushing buttons. Players not wanting the hassle aren't obliged to push the button, but those who want some direction or a challenge get some at the price of having to  do the thing

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
That would definitely be an improvement over what we have now, but it still suffers the problem that our current objectives, optional or not, are pretty uninteresting when followed to the letter

**Comment by kilozombie** [comment]
I would be down for writing up a basic outline on possible objectives following the criteria given by @I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE in that long post up there, if that's where a lot of the support is. Not really qualified to code the framework, though

**Comment by nuklearcellphoneg** [comment]
Regardless of how open-ended the objectives are, having them will inevitably lead to people at round-end complain: 'ugh, he attacked me, the RD, when it was the CMO who ruined his life, admins enforcing rules on antags when' and so on.

Being quantifiable has had little to no bearing in the past on whether or not someone post-round was salty that someone did or did not do an objective. Classic example would be an antag with an emag being attacked with a laser, they fight back and wind up killing multiple people, but they had a Minimize casualties objective. Sure, they are justified, but people will ignore the context. 

Statistics are a step in the right direction. Objectives with specificity, however, will almost always either pigeonhole a player's play style by them attempting to adhere to the objective's guidelines, or bystanders after the fact being of the opinion that the antagonist did not to follow the guidelines closely enough, again regardless of the context. 

'The CMO ruined your life' is an infinitely better objective than 'Steal the jetpack', but there will continue to be idiots as both antags, observers, and bystanders.

Just make it so that at round start they must write and lock-in an objective before accessing their tator PDA, powers, or whatever, that way they have a motif for their round. If you'd argue that someone might just write 'I am CRAZY', that's true, but this same person is going to ignore the 'The CMO ruined your life' objective anyway, might as well let their objective stand as a testament to how unoriginal and incapable of arr pee they are.

**Comment by kilozombie** [comment]
How about several options? Maybe three given motivations like 'The CMO ruined your life', and a blank slot where you can write in whatever you want? Sort of how like you don't _need_ a writing prompt given to you to start writing, but they can be good platforms to develop your own ideas.

**Comment by nuklearcellphoneg** [comment]
Give them the option of accepting a guideline/objective like that, but allow for them to opt out/write their own. Objectives have been problematic because people either follow them too closely, causing originality to flounder(wow he just stole a jet pack and hid in the locker the whole three hour round!!!!) or after the fact, because those who were not the antag judged them(subjectively LMAO) for not following the objective objectively enough(LMAO). Again, see Minimize Casualties guy having to kill people who were trying to kill him, and so was justified, and people get mad because he didn't die for his <s>sins</s> objective.

Give a sizeable(1,000 would be nice, but at least a few hundred or else people will meme the objectives in green text stories(ANOTHER 'THE CMO RUINED MY LIFE' ROUND AGAIN etc) number of objectives to choose from, allow people to create their own motif and forego the list entirely(in case their gimmick does not fit the list), or allow them to ignore objectives altogether(regardless of how interesting the objective is or if the antag went into the round with the intention of fulfilling their homemade objective, it's possible they want to just improvise, or not be beholden to it after the fact, because of muh emergent gameplay, see Minimize Casualties guy).

This way, people who did not expect to be antag can have inspiration if they want it, people who want to have a gimmick can, at the end of the round, point at the motif they picked at the beginning and say 'see that was what I was going for the whole time', and the people who just want to go with the flow can do just that, and not worry about having to deal with people who were unhappy with them not following the objective,

**Comment by kilozombie** [comment]
Having the entire list all at once is probably too much, if that's what you mean. I was thinking a handful selected at random, and maybe a button to reroll as many times as you want, as well as being able to pick several overall. A lot of objectives would have many variants (different people, different departments, etc. etc.) but there should definitely be way more variety than currently exists.

**Comment by kilozombie** [comment]
For what it's worth, that is still absolutely identical to what @I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE  recommended, and it's what I'm going to start drawing out some prompts for.

**Comment by Orangerush** [comment]
Just remove all objectives and replace them with a short story for each type of antag.
Just beeing urself was good if someone makes it not sound like that much of a joke

**Comment by kilozombie** [comment]
I wrote this up as both a lore prompt and general outline of the idea--

![image](https://user-images.githubusercontent.com/6062247/44753212-2bd47c80-aad2-11e8-8583-0691edae847b.png)

Still definitely a draft, but provides justification for all manners of playing Traitor.

**Comment by nuklearcellphoneg** [comment]
>Having the entire list all at once is probably too much, 

No, I mean rerolling through a number of premade guidelines/objectives as food for thought, so someone who's not sure what they want to do can snap their fingers and say I'll go with that. I was implying that it would be important to have a number of these so it wouldn't get stale, or so that there wouldn't be one that people repeatedly try to roll, so on.

However, it's _**equally**_ important also for them to be able to make up their own(so they can run a gimmick/motif) or just turn the objectives off entirely(I don't want to deal with it/I don't want people to judge me afterwards if they think I did it 'wrong' or I didn't manage it).

The lore prompt is meh, it's something that people will just exit out of immediately, or read it and soon be something they have to exit out of immediately to get on with things every round, or scroll past, or scroll around seeing whether their code is before or after the lore prompt(How to tell a new antag 101: At roundstart they spend five minutes sitting in one spot then pull out their PDA). It's a whole lot of fluff that a newer player must read over and comprehend and wonder if there is an important part in here they need to pay attention to, and an older player has to skip, when the crux is restated several times; 'do what they wish', 'there was no longer a question of asking them to do their 'objectives'', 'these new traitors would make their own objectives'

 Simply put, you're going from one extreme to the other:

Just bee urself

vs

Let me tell you thirty different ways you can do whatever you want and also why you can do whatever you want and about how back in my day we weren't supposed to necessarily do what we wanted but we did what we wanted anyway, so now you can just do what you want and be yourself.

**Comment by kilozombie** [comment]
The lore prompt isn't even meant to be handed to the players at any point. I don't see the sense in that. It's for the wiki, _maybe_, but definitely not anywhere ingame. It's lore, my dude. You can sum it up in about twenty words for ingame.

Everything else you're saying has already been established as something I agree with, like having lots of objectives and being able to write in your own. Not sure who you're trying to convince at this point.

**Comment by nuklearcellphoneg** [comment]
When you said prompt, it made me think of a game mechanic prompt, like how it prompts you for a clown or silicon name, or, relevant to this discussion, how the game would prompt you to pick/roll an objective, write out your own objective, or forgoing an objective altogether.

**Comment by MadmanMartian** [comment]
While I do think this is an interesting way of approaching the game, could we not make changes to objectives until after role datums are finished?

**Comment by kilozombie** [comment]
I am personally fine with waiting, but that's just me. I do think that this would probably be made _easier_ once role datums are finished, but... it's also possible that some of this could be designed into how role datums work?

**Comment by Hoshi-chan** [comment]
Murderbones aren't the problem with objectives, inactive antags and people who chase after greentext, then do nothing, are the problem with objectives. 

We don't need less objectives, we need more engaging objectives. That being said, I'm for this change because anything is better than what we have now, and it's not like merging it for a week to see how it goes would hurt anything.

Only legitimate argument against at least trying it is that it makes more work for the people working on datums right now.

**Comment by kilozombie** [comment]
That's just about the consensus right now. I agree with most people in this thread, especially on the front of removing greentext checks. I think @I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE and @yclatious have both said it much better than I can.

**Comment by DeityLink** [comment]
>We don't need less objectives, we need more engaging objectives.

This.
